src/unix/fs.c

```diff
@@ -217,6 +217,88 @@ static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
 }
 
 
+static int uv__fs_mkstemp(uv_fs_t* req) {
+  int r;
+#ifdef UV__O_CLOEXEC
+  int (*mkostemp_function)(char*, int);
+  static int no_cloexec_support;
+#endif
+  char *path;
+  int path_length;
+
+  path = (char*) req->path;
```

Comment:
I'm thinking that maybe we should change back `uv_fs_t.path` to a `char*`.  We can do that without API or ABI breakage.  In-place modification of a `const char*` is surprising, to say the least.

Also, I expect you need to update `uv_fs_req_cleanup()` to special-case `UV_FS_MKSTEMP` requests.